sdk/storage/azure-storage-blob-v2/src/main/java/com/azure/v2/storage/blob/ContainerClient.java

```diff
@@ -0,0 +1,1065 @@
+            sourceLeaseId, requestOptions);
+    }
+
+    /**
+     * Renames an existing container.
+     * 
+     * @param containerName The container name.
+     * @param sourceContainerName Required. Specifies the name of the container to rename.
+     * @param timeout The timeout parameter is expressed in seconds. For more information, see &lt;a
+     * href="https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/setting-timeouts-for-blob-service-operations"&gt;Setting
+     * Timeouts for Blob Service Operations.&lt;/a&gt;.
+     * @param requestId Provides a client-generated, opaque value with a 1 KB character limit that is recorded in the
+     * analytics logs when storage analytics logging is enabled.
+     * @param sourceLeaseId A lease ID for the source path. If specified, the source path must have an active lease and
+     * the lease ID must match.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws StorageErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    @Metadata(generated = true)
+    public void rename(String containerName, String sourceContainerName, Integer timeout, String requestId,
+        String sourceLeaseId) {
+        this.serviceClient.rename(containerName, sourceContainerName, timeout, requestId, sourceLeaseId);
+    }
+
+    /**
+     * The Batch operation allows multiple API calls to be embedded into a single HTTP request.
+     * 
+     * @param containerName The container name.
+     * @param contentLength The length of the request.
+     * @param multipartContentType Required. The value of this header must be multipart/mixed with a batch boundary.
+     * Example header value: multipart/mixed; boundary=batch_&lt;GUID&gt;.
+     * @param body Initial data.
+     * @param timeout The timeout parameter is expressed in seconds. For more information, see &lt;a
+     * href="https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/setting-timeouts-for-blob-service-operations"&gt;Setting
+     * Timeouts for Blob Service Operations.&lt;/a&gt;.
+     * @param requestId Provides a client-generated, opaque value with a 1 KB character limit that is recorded in the
+     * analytics logs when storage analytics logging is enabled.
+     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws StorageErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the response.
+     */
+    @Metadata(generated = true)
+    public Response<InputStream> submitBatchWithResponse(String containerName, long contentLength,
```

Comment:
Something that we can save for another PR, but we should probably make a pass on hiding APIs that we do not want to expose in the main blob package. For example, the batch APIs have their own batch specific package right now outside of the main blob package and so we probably do not want to include them by default until we have made a decision on how we want to expose the batch API in vNext.

Reply:
Adding a note here. Typespec has the ability to hide operations. Should be a matter of updating the model to hide it.